comp/remote-config/rcclient/rcclient.go

```diff
@@ -0,0 +1,121 @@
+package rcclient
+
+import (
+	"time"
+
+	"github.com/pkg/errors"
+	"go.uber.org/fx"
+
+	"github.com/DataDog/datadog-agent/pkg/config/remote"
+	"github.com/DataDog/datadog-agent/pkg/config/remote/data"
+	"github.com/DataDog/datadog-agent/pkg/remoteconfig/state"
+	"github.com/DataDog/datadog-agent/pkg/util/fxutil"
+	"github.com/DataDog/datadog-agent/pkg/version"
+)
+
+// Component is the component type.
+type Component interface {
+	// Start the remote config client to listen to AGENT_TASK configurations
+	Listen() error
+}
+
+// RCListener is the FX-compatible listener, so RC can push updates through it
+type RCListener func(task state.AgentTaskConfig) (bool, error)
+
+type rcClient struct {
+	client        *remote.Client
+	taskProcessed map[string]bool
+
+	listeners []RCListener
+}
+
+type dependencies struct {
+	fx.In
+
+	Listeners []RCListener `group:"rCListener"` // <-- Fill automatically by Fx
+}
+
+// Module defines the fx options for this component.
+var Module = fxutil.Component(
+	fx.Provide(newRemoteConfig),
+)
+
+func newRemoteConfig(deps dependencies) (Component, error) {
+	rc := rcClient{
+		listeners: deps.Listeners,
+		client:    nil,
+	}
+
+	return rc, nil
+}
+
+// Listen start the remote config client to listen to AGENT_TASK configurations
+func (rc rcClient) Listen() error {
+	c, err := remote.NewUnverifiedGRPCClient(
+		"core-agent", version.AgentVersion, []data.Product{data.ProductAgentTask}, 1*time.Second,
+	)
+	if err != nil {
+		return err
+	}
+
+	rc.client = c
+	rc.taskProcessed = map[string]bool{}
+
+	rc.client.RegisterAgentTaskUpdate(rc.agentTaskUpdateCallback)
+
+	rc.client.Start()
+
+	return nil
+}
+
+// agentTaskUpdateCallback is the callback function called when there is an AGENT_TASK config update
+// The RCClient can directly call back listeners, because there would be no way to send back
+// RCTE2 configuration applied state to RC backend.
+func (rc rcClient) agentTaskUpdateCallback(updates map[string]state.AgentTaskConfig) {
+	for configPath, c := range updates {
+		// Check that the flare task wasn't already processed
+		if !rc.taskProcessed[c.Config.UUID] {
+			rc.taskProcessed[c.Config.UUID] = true
```

Comment:
Will you have a limited number of events? I am asking because `rc.taskProcessed` grows but it is never release.

Reply:
Since we only support flares, we should only receive one entry at every flare request
It shouldn't happen often

If we ever support more tasks we'll  switch to a map with timeout